src/unstated.js

```diff
@@ -36,23 +34,12 @@ export class Container<State: {}> {
 
       this.state = Object.assign({}, this.state, nextState);
 
-      let completed = 0;
-      let total = this._listeners.length;
+      let promises = this._listeners.map(listener => listener());
 
-      this._listeners.forEach(fn => {
-        if (!callback) {
-          fn();
-          return;
+      return Promise.all(promises).then(() => {
+        if (callback) {
+          return callback();
```

Comment:
If something inside the callback function throws then the error might get swallowed.